evaluation/red_teaming.py

```diff
@@ -27,48 +30,70 @@ async def run_red_teaming(
     config: dict,
     red_teaming_llm: PromptChatTarget,
     prompt_target: PromptChatTarget,
+    max_turns: int,
+    compare: bool,
 ):
     """Run red teaming attack with provided scorers using Red Teaming Orchestrator."""
+    prompt_target_list = []
+    if compare and isinstance(prompt_target, AppChatTarget):
+        compared_models = config.get("models")
+        for compare_model in compared_models:
+            prompt_target_copy = copy.copy(prompt_target)
+            prompt_target_copy.target_parameters = copy.deepcopy(prompt_target.target_parameters)
+            prompt_target_copy.target_parameters["overrides"]["set_model"] = compare_model
+            prompt_target_list.append(prompt_target_copy)
+    elif isinstance(prompt_target, AppChatTarget):
+        # Default use the first model in the config
+        prompt_target.target_parameters["overrides"]["set_model"] = config.get("models")[0]
+        prompt_target_list.append(prompt_target)
+    else:
+        prompt_target_list.append(prompt_target)
     logger.info("Running red teaming attack, with scorers from '%s'", scorer_dir)
     scorers = [Path(scorer_file) for scorer_file in glob.glob(os.path.join(scorer_dir, "*.yaml"))]
     # Add built-in scorers
     scorers.extend([path.value for path in TrueFalseQuestionPaths])
-    results = []
-
-    for scorer_path in scorers:
-        logger.info("Runing red teaming with scorer YAML: %s", scorer_path)
-        try:
-            scorer_data = validate_scorer_yaml(scorer_path)
-        except ValueError as e:
-            logger.error(f"Invalid scorer YAML: {e}")
-            continue
-
-        scorer = SelfAskTrueFalseScorer(chat_target=red_teaming_llm, true_false_question_path=scorer_path)
-        attack_strategy = AttackStrategy(
-            strategy=text_generation_strategy_path,
-            conversation_objective=scorer_data.get("conversation_objective", ""),
-        )
-
-        with RedTeamingOrchestrator(
-            attack_strategy=attack_strategy,
-            red_teaming_chat=red_teaming_llm,
-            prompt_target=prompt_target,
-            initial_red_teaming_prompt="how can I help you?",
-            scorer=scorer,
-            verbose=True,
-        ) as red_teaming_orchestrator:
-            score = await red_teaming_orchestrator.apply_attack_strategy_until_completion_async(max_turns=3)
-            results.append(score)
+
+    results_for_all_models = {}
+    for prompt_target in prompt_target_list:
+        if isinstance(prompt_target, AppChatTarget):
+            model_name = prompt_target.target_parameters["overrides"]["set_model"]
+        else:
+            model_name = prompt_target.__class__.__name__
```

Comment:
What's happening here?

Reply:
> What's happening here?

The way of getting the model name when we compare only LLMs is different from when comparing between the entire applications because the targets are set up in different ways,.

Reply:
Absolutely, I was just wondering whether using 2 under attributes is the only way to fetch the model name!